api/ocs/__init__.py

```diff
@@ -320,134 +405,280 @@ def parse_atoms(sequence: List[dict], qa_states: List[QAState]) -> List[Atom]:
-                
-        qa_states = [QAState[log_entry[OCSProgramProvider._ObsKeys.QASTATE].upper()] for log_entry in data[OCSProgramProvider._ObsKeys.LOG]]
-
-        site = Site.GN if data[OCSProgramProvider._ObsKeys.ID].split('-')[0] == 'GN' else Site.GS
-        status = ObservationStatus[data[OCSProgramProvider._ObsKeys.STATUS]]
-        priority = Priority.HIGH if data[OCSProgramProvider._ObsKeys.PRIORITY] == 'HIGH' else (Priority.LOW if data[OCSProgramProvider._ObsKeys.PRIORITY] == 'LOW' else Priority.MEDIUM)
-        atoms = OCSProgramProvider.parse_atoms(data[OCSProgramProvider._ObsKeys.SEQUENCE], qa_states)
- 
-        obs = Observation(data[OCSProgramProvider._ObsKeys.ID],
-                          data[OCSProgramProvider._ObsKeys.INTERNAL_ID],
-                          int(name.split('-')[1]),
-                          data[OCSProgramProvider._ObsKeys.TITLE],
-                          site,
-                          status,
-                          True if data[OCSProgramProvider._ObsKeys.PHASE2] != 'Inactive' else False,
-                          priority,
-                          None,
-                          SetupTimeType[data[OCSProgramProvider._ObsKeys.SETUPTIME_TYPE]],
-                          timedelta(milliseconds=data[OCSProgramProvider._ObsKeys.SETUPTIME]),
-                          None,
-                          None,
-                          None,
-                          OCSProgramProvider.obs_classes[data[OCSProgramProvider._ObsKeys.OBS_CLASS]],
-                          targets,
-                          guide_stars,
-                          atoms,
-                          constraints,
-                          None)
-        obs.exec_time = sum([atom.exec_time for atom in atoms], timedelta()) + obs.acq_overhead
-
-        return obs
-    
+
     @staticmethod
-    def parse_time_allocation(json: dict) -> TimeAllocation:
-        return TimeAllocation(TimeAccountingCode(json[OCSProgramProvider._TAKeys.CATEGORIES][0][OCSProgramProvider._TAKeys.CATEGORY]),
-                              timedelta(milliseconds=json[OCSProgramProvider._TAKeys.AWARDED_TIME]),
-                              timedelta(milliseconds=0),
-                              timedelta(milliseconds=json[OCSProgramProvider._TAKeys.CATEGORIES][0][OCSProgramProvider._TAKeys.PROGRAM_TIME]),
-                              timedelta(milliseconds=json[OCSProgramProvider._TAKeys.CATEGORIES][0][OCSProgramProvider._TAKeys.PARTNER_TIME]))
+    def parse_target(data: dict) -> Target:
+        """
+        Parse a general target - either sidereal or nonsidereal - from the supplied data.
+        """
+        tag = data[OcsProgramProvider._TargetKeys.TAG]
+        if tag == 'sidereal':
+            return OcsProgramProvider.parse_sidereal_target(data)
+        elif tag == 'nonsidereal':
+            return OcsProgramProvider.parse_sidereal_target(data)
```

Comment:
This should be non-sidereal I imagine?

Reply:
Ooops! You are right. Good catch. Thanks!